pkg/macvlan/macvlan.go

```diff
@@ -314,6 +315,34 @@ func macvlanCmdAdd(args *skel.CmdArgs) error {
 	}
 
 	err = netns.Do(func(_ ns.NetNS) error {
+		// Get default interface
+		existingLink, err := netlink.LinkByName("eth0")
+		if err != nil {
+			return fmt.Errorf("couldn't get interface eth0: %v", err)
+		}
+
+		// Delete default route
+		existingDefaultRoute := netlink.Route{
+			LinkIndex: existingLink.Attrs().Index,
+			Dst: nil,
+		}
+		netlink.RouteDel(&existingDefaultRoute)
+
+		// Get macvlan interface
+		macvlanLink, err := netlink.LinkByName(args.IfName)
+		if err != nil {
+			return fmt.Errorf("could not get interface: %v", err)
+		}
```

Comment:
Okay so let's suppose we delete the eth0 default route, and we come to this part and can't get the macvlan interface. Then we are busted right ? Shouldn't we check for the existence of both the interfaces before we start on this logic ?

Reply:
I guess this was moved due to the new push, but there's already code for that in `netlink.LinkByName(args.IfName)`

Reply:
> I guess this was moved due to the new push, but there's already code for that in `netlink.LinkByName(args.IfName)`

yea this was a problem in your old version. saw that you fixed it in the current version (will take a look)